packages/@ama-sdk/schematics/schematics/typescript/core/openapi-codegen-typescript/src/main/java/com/amadeus/codegen/ts/AbstractTypeScriptClientCodegen.java

```diff
@@ -512,11 +506,20 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         + "list container.");
     }
 
-    // If the x-date-timezone is present in the specification, replace utils.Date with Date
-    if (("utils.Date".equals(property.dataType) || "utils.DateTime".equals(property.dataType)) && property.vendorExtensions.containsKey("x-date-timezone")) {
-      property.dataType = "Date";
-      property.datatypeWithEnum = "Date";
-      property.baseType = "Date";
+    if (property.vendorExtensions.containsKey("x-date-timezone")) {
+      throw new IllegalArgumentException("'x-date-timezone' is deprecated and conflicts with the 'x-local-timezone' vendor." +
```

Comment:
Shouldn't we throw if both `x-date-timezone` and `x-local-timezone` are present?
And maybe log a non-blocking warning if `x-date-timezone` is there?

Reply:
I think we were to completely drop the x-date-timezone once we started supporting the x-local-timezone.

Reply:
As discussed, we need the information of the x-vendor used in the properties in order to set the property original value. This is something we cannot really do just like this (or I guess we would need to rework all the properties as soon as we get the info? This does not seem such a good idea and I am not sure we can pull it off).

I followed your advice and put the migration guide in the error message to help our user to migrate.